### DISCUSSION TITLE
请教有办法Mixin一个对象而不是一个Class么？

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1465] [Q&A]
现在Mixin是作用在Class类上的，意味着一定是由TS来Spawn产生的对象才能Mixin。但很多对象可能是由UE的其他机制创建和销毁的，例如ListView中的一个Entry等等。
这些对象初始化的时候，例如蓝图的BeginPlay，我能在对象级别上去Mixin一个JS类的对象进去么？

**Comment by chexiongsheng** [comment]
没办法。C++和蓝图就没办法针对对象修改方法。
拿到对象了不就可以通过对象获取UClass了么？为什么还要mixin对象？

**Comment by watsonsong** [comment]
主要是试用了引擎类继承，感觉大量的变量和函数需要认真加@noblueprint，而且没办法使用默认的构造函数。往往写法模式上会变成引擎类继承只是一个壳，用于定义例如可同步属性和RPC等，具体逻辑代理给另外一个纯JS的类实现功能。引擎类继承在代码重构，换名字的时候替换起来也会比较麻烦。
但如果这样，那不如这个壳直接就做成一个真正的蓝图还更加直接一些。因此在考虑，要么蓝图能配置一个脚本对象去mixin它。或者蓝图持有一个JSObject，然后K2Node开发一个能蓝图调用JSObject的节点。
后者看起来得开发一下，就想问问如果对象级别能mixin的话基本上也能得到类似的效果，可能还会更方便。
